src/runtime/security/server_negotiation.cpp

```diff
@@ -33,5 +36,38 @@ void server_negotiation::start()
     ddebug_f("{}: start negotiation", _name);
 }
 
+void server_negotiation::handle_request(negotiation_rpc rpc)
+{
+    if (_status == negotiation_status::type::SASL_LIST_MECHANISMS) {
+        on_list_mechanisms(rpc);
+        return;
+    }
+}
+
+void server_negotiation::on_list_mechanisms(negotiation_rpc rpc)
+{
+    if (rpc.request().status == negotiation_status::type::SASL_LIST_MECHANISMS) {
+        std::string mech_list = boost::join(supported_mechanisms, ",");
+        ddebug_f("{}: reply server mechs({})", _name, mech_list);
```

Comment:
```suggestion
        std::string mech_list = boost::join(supported_mechanisms, ",");
        ddebug_f("{}: reply server mechs({})", _name, mech_list);
```

This log will be as many as client connections. I don't think it's necessary.